java/gateway-core/src/test/java/com/avaulta/gateway/rules/ParameterSchemaUtilsTest.java

```diff
@@ -0,0 +1,86 @@
+package com.avaulta.gateway.rules;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ParameterSchemaUtilsTest {
+
+
+    String EXAMPLE_REVERSIBLE = "p~nVPSMYD7ZO_ptGIMJ65TAFo5_vVVQQ2af5Bfg7bW0Jq9JIOXfBWhts_zA5Ns0r4m";
+
+
+    ParameterSchemaUtils parameterSchemaUtils = new ParameterSchemaUtils();
+    @Test
+    public void validation_reversiblePseudonym() {
+        ParameterSchema parameterSchema = ParameterSchema.builder()
+                .type(ParameterSchema.ValueType.STRING.getEncoding())
+                .format(ParameterSchema.StringFormat.REVERSIBLE_PSEUDONYM.getStringEncoding())
+                .build();
+
+        assertTrue(parameterSchemaUtils.validate(parameterSchema, EXAMPLE_REVERSIBLE));
+        assertFalse(parameterSchemaUtils.validate(parameterSchema, "not-a-pseudonym"));
+        assertTrue(parameterSchemaUtils.validate(parameterSchema, null));
+    }
+
+
+    @Test
+    public void validation_enum() {
+        ParameterSchema parameterSchema = ParameterSchema.builder()
+                .type(ParameterSchema.ValueType.STRING.getEncoding())
+                .enumValues(Arrays.asList("v0.1", "beta"))
+                .build();
+
+        assertTrue(parameterSchemaUtils.validate(parameterSchema, "beta"));
+        assertTrue(parameterSchemaUtils.validate(parameterSchema, "v0.1"));
+        assertFalse(parameterSchemaUtils.validate(parameterSchema, "v0.2"));
+
+        assertTrue(parameterSchemaUtils.validate(parameterSchema, null));
```

Comment:
this could also be an option, absence allowed or not. If we are expecting an enum, I don't see why null should be accepted unless it is not mandatory.
Same above for the reversible.